odk1-src/form-logic.rst

```diff
@@ -639,9 +617,55 @@ The :th:`repeat_count` column can reference
   integer, child_age, Child's age,
   end_repeat, , , 
 
+.. _repeat_based_on_condition:
+
+Repeating as long as a condition is met
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the enumerator won't know how many repetitions are needed ahead of time, you can still avoid the "Add new group?" dialog by using the answer to a question to decide whether another repeat instance should be added. In the example below, repeated questions about plants will be asked as long as the user answers "yes" to the last question.
 
-.. seealso:: :doc:`form-repeats`
+.. csv-table:: survey
+  :header: type, name, label, calculation, repeat_count
   
+  calculate, count, , count(${plant})
+  begin_repeat, plant, Plant, , if(${count} = 0 or ${plant}[position()=${count}]/more_plants = 'yes', ${count} + 1, ${count})
+  text, species, Species,
+  integer, estimated_size, Estimated size,
+  select_one yes_no, more_plants, Are there more plants in this area?,
+  end_repeat, , , ,
+
+.. csv-table:: choices
+  :header: list_name, name, label
+  
+  yes_no, yes, Yes
+  yes_no, no, No
+
+This works by maintaining a :func:`count` of the existing repetitions and either making :th:`repeat_count` one more than that if the continuing condition is met or keeping the :th:`repeat_count` the same if the ending condition is met. 
+
+In the `repeat_count` expression, `${count} = 0` ensures that there is always at least one repeat instance created. The continuing condition is `${plant}[position()=${count}]/more_plants = 'yes'` which means "the answer to `more_plants` was `yes` the last time it was asked." The expression `position()=${count}` uses the :func:`position` function to select the last plant that was added. Adding `/more_plants` to the end of that selects the `more_plants` question.
+
+.. _zero-repetitions:
+
+Representing zero or more repetitions
```

Comment:
Maybe just "Zero or more repeptitions" is a bit simpler?

Reply:
How about "Repeating zero or more times" to match the previous subheading?

Reply:
👍